base_update_restrict_model/views/res_users_views.xml

```diff
@@ -0,0 +1,33 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+
+    <record id="view_users_form" model="ir.ui.view">
+        <field name="name">res.users.form</field>
+        <field name="model">res.users</field>
+        <field name="inherit_id" ref="base.view_users_form" />
+        <field name="arch" type="xml">
+            <field name="company_id" position="after">
+                <field name="restrict_update_permit" invisible="1" />
+            </field>
+            <div class="oe_button_box" name="button_box" position="inside">
+                <button
+                    name="set_restrict_update_permit"
+                    type="object"
+                    class="oe_stat_button"
+                    icon="fa-eye"
```

Comment:
These icons are a bit misleading (making an object read-only and restricting updates are different things).

I also prefer this style: https://github.com/odoo/odoo/blob/2fb44a86819fc6a7a6507d120c79639e2e9f8a6b/odoo/addons/base/views/res_users_views.xml#L166-L168